src/client.cpp

```diff
@@ -324,12 +304,70 @@ PVR_ERROR GetChannelGroupMembers(ADDON_HANDLE handle, const PVR_CHANNEL_GROUP& g
 
 PVR_ERROR SignalStatus(PVR_SIGNAL_STATUS& signalStatus)
 {
-  snprintf(signalStatus.strAdapterName, sizeof(signalStatus.strAdapterName), "IPTV Simple Adapter 1");
-  snprintf(signalStatus.strAdapterStatus, sizeof(signalStatus.strAdapterStatus), "OK");
+  strncpy(signalStatus.strAdapterName, "IPTV Simple Adapter 1", sizeof(signalStatus.strAdapterName) - 1);
+  strncpy(signalStatus.strAdapterStatus, "OK", sizeof(signalStatus.strAdapterStatus) - 1);
 
   return PVR_ERROR_NO_ERROR;
 }
 
+PVR_ERROR IsEPGTagPlayable(const EPG_TAG* tag, bool* bIsPlayable)
+{
+  if (!m_data)
+    return PVR_ERROR_SERVER_ERROR;
+  if (!settings.IsCatchupEnabled())
+    return PVR_ERROR_NOT_IMPLEMENTED;
+
+  const time_t now = time(nullptr);
+  Channel channel;
+
+  // Get the channel and set the current tag on it if found
+  *bIsPlayable = m_data->GetChannel(static_cast<int>(tag->iUniqueChannelId), channel);
+
+  *bIsPlayable = *bIsPlayable &&
+                 settings.IsCatchupEnabled() &&
+                 tag->startTime < now &&
+                 channel.IsCatchupSupported() &&
+                 tag->startTime >= (now - static_cast<time_t>(channel.GetCatchupDaysInSeconds())) &&
+                 (!settings.CatchupOnlyOnFinishedProgrammes() || tag->endTime < now);
+
+  return PVR_ERROR_NO_ERROR;
+}
+
+PVR_ERROR GetEPGTagStreamProperties(const EPG_TAG* tag, PVR_NAMED_VALUE* properties, unsigned int* iPropertiesCount)
+{
+  Logger::Log(LEVEL_DEBUG, "%s - Tag startTime: %ld \tendTime: %ld", __FUNCTION__, tag->startTime, tag->endTime);
+
+  if (!m_data || !tag || !properties || !iPropertiesCount)
+    return PVR_ERROR_SERVER_ERROR;
+
+  if (*iPropertiesCount < 1)
+    return PVR_ERROR_INVALID_PARAMETERS;
+
+  if (m_data->GetChannel(static_cast<int>(tag->iUniqueChannelId), m_currentChannel))
+  {
+    unsigned int propertiesMax = *iPropertiesCount;
+    *iPropertiesCount = 0;
+    Logger::Log(LEVEL_DEBUG, "%s - GetPlayEpgAsLive is %s", __FUNCTION__, settings.CatchupPlayEpgAsLive() ? "enabled" : "disabled");
+
+    std::map<std::string, std::string> catchupProperties;
+    if (settings.CatchupPlayEpgAsLive())
+      m_catchupController->ProcessEPGTagForTimeshiftedPlayback(*tag, m_currentChannel, catchupProperties);
+    else
+      m_catchupController->ProcessEPGTagForVideoPlayback(*tag, m_currentChannel, catchupProperties);
+
+    std::string catchupUrl = m_catchupController->GetCatchupUrl(m_currentChannel);
```

Comment:
`const std::string`? Or is it modified by `StreamUtils::SetAllStreamProperties`?